### PR TITLE
feat(atomic,headless): add atomic-insight-timeframe-facet

### PR DESCRIPTION
**Pull request by liviarett**
https://user-images.githubusercontent.com/30511433/182891289-5ea9e5fb-8566-4bc7-b054-8fb3b3f6e9d6.mov

### CODE REVIEW

File: packages/atomic/src/components/common/atomic-timeframe/atomic-timeframe.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import {Component, Prop} from '@stencil/core';
 import {RelativeDateUnit} from '@coveo/headless';
-import {Timeframe} from './timeframe';
-
+import {InsightTimeframe, Timeframe} from '../facets/timeframe-facet-common';
+// TODO: move
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/common/atomic-timeframe/atomic-timeframe.tsx:4]: Move where?

File: packages/atomic/src/components/insight/atomic-insight-timeframe-facet/atomic-insight-timeframe-facet.tsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+   * Minimum: `0`
+   * Default: `1000`
+   */
+  @Prop({reflect: true}) public injectionDepth = 1000;
+
+  /**
+   * The required facets and values for this facet to be displayed.
+   * Examples:
+   * ```html
+   * <atomic-facet facet-id="abc" field="objecttype" ...></atomic-facet>
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/insight/atomic-insight-timeframe-facet/atomic-insight-timeframe-facet.tsx:133]:
```suggestion
   * <atomic-insight-facet facet-id="abc" field="objecttype" ...></atomic-insight-facet>
```

**Review by ThibodeauJF — APPROVED** [review]
The timeframe facet is my personal favorite 😊

File: packages/atomic/src/components/common/atomic-timeframe/atomic-timeframe.tsx (hunk 1 above)
Comment (liviarett) [reply packages/atomic/src/components/common/atomic-timeframe/atomic-timeframe.tsx:4]: 🤷‍♀️

**Review by aelfaleh — COMMENTED** [review]
Did you succeed  to create the component with formatted `timeframe`: such as `Past week` ?

**Comment by nathanlb** [comment]
Nice video! The atomic insight panel is looking really awesome! I'm a bit confused by the current changeset though. It's rather large and seems to include changes to many components beyond the timeframe facet. Is it as intended?

File: packages/atomic/src/components/insight/atomic-insight-numeric-facet/atomic-insight-numeric-facet.tsx [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+   * Minimum: `0`
+   * Default: `1000`
+   */
+  @Prop({reflect: true}) public injectionDepth = 1000;
+
+  /**
+   * The required facets and values for this facet to be displayed.
+   * Examples:
+   * ```html
+   * <atomic-facet facet-id="abc" field="objecttype" ...></atomic-facet>
```
Comment (btaillon) [inline packages/atomic/src/components/insight/atomic-insight-numeric-facet/atomic-insight-numeric-facet.tsx:169]: I would demo an insight facet here too

**Comment by liviarett** [comment]
@nathanlb the diff is correct now!

**Comment by liviarett** [comment]
@aelfaleh thanks for flagging that, I hadn't even noticed that it wasn't doing what it's supposed to do. after some debugging, I noticed that the response coming from the insight isn't mapped like the search interface one is, so I added that to the latest commit.
I'd appreciate it if you guys could have a look at it and approve, please!

**Review by aelfaleh — APPROVED** [review]
LGTM

**Pull request merged**